src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -279,8 +279,26 @@ private void RaiseLockFileTargets()
 
         private void GetPackageAndFileDependencies(LockFileTarget target)
         {
-            var resolvedPackageVersions = target.Libraries
-                .ToDictionary(pkg => pkg.Name, pkg => pkg.Version.ToNormalizedString(), StringComparer.OrdinalIgnoreCase);
+            // In case of there are duplicated packages in libraries with differing versions,
+            // We skip package altogether
+            var resolvedPackageVersions = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            var duplicatedPackages = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (LockFileTargetLibrary library in target.Libraries)
+            {
+                string libraryName = library.Name;
+                if (duplicatedPackages.Contains(libraryName))
+                {
+                    continue;
```

Comment:
Add some logging so that i.e. a binlog shows that duplicate packages are consumed?

Reply:
Can you point me to some example of logging, that would be helpful.